client/apps/edit/components/content/section_list/index.jsx

```diff
@@ -1,41 +1,52 @@
 import PropTypes from 'prop-types'
 import React, { Component } from 'react'
+import { bindActionCreators } from 'redux'
+import { connect } from 'react-redux'
+import * as Actions from 'client/actions/editActions'
 import DragContainer from 'client/components/drag_drop/index.coffee'
 import { SectionContainer } from '../section_container/index'
 import { SectionTool } from '../section_tool/index'
 
 export class SectionList extends Component {
   static propTypes = {
+    actions: PropTypes.object,
     article: PropTypes.object.isRequired,
     channel: PropTypes.object.isRequired,
+    edit: PropTypes.object,
     sections: PropTypes.object.isRequired
   }
 
-  state = {
-    editingIndex: null
-  }
-
   componentDidMount = () => {
     const { sections } = this.props
 
     sections.on('add', this.onNewSection)
   }
 
   onNewSection = (section) => {
+    const { actions, sections } = this.props
+    const activeSection = sections.indexOf(section)
+
+    actions.changeSection(activeSection)
+  }
+
+  onDragEnd = (newSections) => {
     const { sections } = this.props
 
-    const editingIndex = sections.indexOf(section)
-    this.setState({ editingIndex })
+    sections.reset(newSections)
+    // force update until sections live in state
+    this.setState({lastUpdated: new Date()})
```

Comment:
Backbone events do not trigger re-render in connected components, so need to do it manually for now.  Long-run goal is moving all the backbone actions into redux.

Reply:
Maybe add `TODO` to the comment so that it jumps out in editor?